### PR TITLE
[DEVX-2783] Add publish command

### PR DESCRIPTION
**Pull request by sasha370**
[DEVX-2783](https://toptal-core.atlassian.net/browse/DEVX-2783)

### Description
We want to build and publish packages in an automated way when we need it.
For now, we only want to do it manually (workflow_dispatch) and not for every merge to the `master` branch.

### How it works

1) Go [here](https://github.com/toptal/xene/actions/workflows/publish_packages.yaml)
2) Press `Run workflow` button
3) Choose the level of version to bump (This will affect all packages that have had changes since the last tag.)
4) Choose a branch to be published (default 'master')
5) Press `Run workflow` button
6) Check that NPM has  new versions of packages

### How to test

1) There were [changes](https://github.com/toptal/xene/pull/41) that were not included in any `version`/tag
2) I run [workflow_dispatch](https://github.com/toptal/xene/actions/runs/6321391701/job/17165341502) with branch 'master'
image

3) It creates a new tag only for the particular package 
image

4) It published that package on [NPM](https://www.npmjs.com/package/@xene/test?activeTab=versions)
The new version  is 0.3.3
image

image

___
### Negative test
1) What if someone tries to `republish` the version?
Then Lerna will throw an [error](https://github.com/toptal/xene/actions/runs/6321741848/job/17166210018#step:12:31) 

image

[DEVX-2783]: https://toptal-core.atlassian.net/browse/DEVX-2783?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/publish_packages.yaml [hunk 1]
```diff
@@ -4,59 +4,67 @@ on:
...
         with:
           fetch-depth: 0
-          ref: ${{ inputs.branch || master }}
+          ref: ${{ inputs.branch || 'master' }}
 
-      - name: GSM Secrets
-        id: secrets_manager
-        uses: toptal/actions/gsm-secrets@main
+      - name: Authenticate to Google Cloud using Workload Identity
+        uses: google-github-actions/auth@v1
```
Comment (sasha370) [inline .github/workflows/publish_packages.yaml:36]: From the public repo we cannot call `/toptal/actions/` so we need to use default actions to get GSM credentials

File: .github/workflows/publish_packages.yaml [hunk 2]
```diff
@@ -4,59 +4,67 @@ on:
...
+        default: 'master'
+      version:
+        description: 'Version to publish (major, minor, patch)'
+        required: true
+        default: 'patch'
+        type: choice
+        options:
+          - 'patch'
+          - 'minor'
+          - 'major'
```
Comment (sasha370) [inline .github/workflows/publish_packages.yaml:18]: indent fix

File: .github/workflows/publish_packages.yaml [hunk 3]
```diff
@@ -4,59 +4,67 @@ on:
...
 
 # Do not allow concurrent runs
 concurrency: ${{ github.workflow }}-${{ github.ref }}
 
 jobs:
   publish:
     name: Publish Packages
-    runs-on: [ 'org/toptal', 'os/linux', 'arch/x64', 'size/standard' ]
+    runs-on: ubuntu-latest
```
Comment (sasha370) [inline .github/workflows/publish_packages.yaml:26]: We cannot use custom runners within `public` repo

**Comment by sasha370** [comment]
@toptal-anvil ping reviewers

File: .github/workflows/publish_packages.yaml [hunk 4]
```diff
@@ -4,59 +4,67 @@ on:
...
+          ref: ${{ inputs.branch || 'master' }}
 
-      - name: GSM Secrets
-        id: secrets_manager
-        uses: toptal/actions/gsm-secrets@main
+      - name: Authenticate to Google Cloud using Workload Identity
+        uses: google-github-actions/auth@v1
         with:
           workload_identity_provider: 'projects/858873486241/locations/global/workloadIdentityPools/gha-pool/providers/github-com'
           service_account: 'gha-xene@toptal-ci.iam.gserviceaccount.com'
```
Comment (ebeagusamuel) [inline .github/workflows/publish_packages.yaml:39]: This is a naive question, but it's safe to use this from a public repo right?

**Review by ebeagusamuel — APPROVED** [review]

**Review by mbie — APPROVED** [review]
Looks promising! Cant wait to see it in action!

**Pull request merged**